medusa/server/web/home/handler.py

```diff
@@ -1293,8 +1293,8 @@ def check_show_for_language(show_obj, language):
         indexer_api_params = indexerApi(show_obj.indexer).api_params.copy()
         indexer_api_params['language'] = language
         indexer_api_params['episodes'] = False
-        t = indexerApi(show_obj.indexer).indexer(**indexer_api_params)
         try:
+            t = indexerApi(show_obj.indexer).indexer(**indexer_api_params)
```

Comment:
Except you're not catching `IndexerUnavailable` in that `try: except`.  You're catching `IndexerShowNotFoundInLanguage`.  Instead you should create the Indexer once (inside a `try: except IndexerUnavailable` and then use that instance throughout.  If `IndexerUnavailable` can occur at any time during the usage of the indexer, then the usage of the function would be the appropriate place to catch `IndexerUnavailable`. 
E.g.
```python
try:
   check_show_for_language(...)
except IndexerUnavailable:
   ...
```